src/main/java/org/casbin/jcasbin/main/SyncedEnforcer.java

```diff
@@ -391,7 +391,7 @@ public List<List<String>> getGroupingPolicy() {
      */
     @Override
     public List<String> getRolesForUser(String name) {
-        return runSynchronized(() -> super.getRolesForUser(name), READ_WRITE_LOCK.readLock());
+        return runSynchronized(() -> super.getRolesForUser(name), READ_WRITE_LOCK.writeLock());
```

Comment:
Because this is a get method, protecting it by write lock will decrease the performance.
I would suggest to change implementation a bit instead.

Reply:
@imp2002

Reply:
If do some make thread safe operator in [Enforcer](https://github.com/casbin/jcasbin/blob/master/src/main/java/org/casbin/jcasbin/main/Enforcer.java), it will decrease the performance of it when use `enforce` not `SyncEnforce`. I don't know a good way to tradeoff. Maybe perform the lock operation on the resource that needs mutual exclusion?